python/MANIFEST.in

```diff
@@ -6,6 +6,11 @@ global-include CMakeLists.txt
 graft pyarrow
 graft cmake_modules
 
+prune benchmarks
+prune cmake_modules
+prune examples
+prune scripts
```

Comment:
This looks good to me but I honestly prefer the changes here: https://github.com/apache/arrow/pull/43325/files 
If we add a new folder or some of those existing ones get removed we don't have to update anything, but we would have to update `MANIFEST.in` every time, so the changes there feel more future proof to me.